pennylane/devices/qubit_mixed/apply_operation.py

```diff
@@ -0,0 +1,641 @@
+        return apply_operation_tensordot(op, state, is_state_batched=is_state_batched)
+
+    # First, flip the left side
+    axis = op.wires[0] + is_state_batched
+    state = _phase_shift(state, axis, phase_factor=1.0j)
+
+    # Second, flip the right side
+    axis = op.wires[0] + is_state_batched + num_wires
+    state = _phase_shift(state, axis, phase_factor=-1.0j)
+
+    return state
+
+
+@apply_operation.register
+def apply_phaseshift(op: qml.PhaseShift, state, is_state_batched: bool = False, debugger=None, **_):
+    """Applies a :class:`~.Phaseshift` operation by multiplying the state by the Phaseshift matrix."""
+    num_wires = int((len(math.shape(state)) - is_state_batched) / 2)
+    n_dim = math.ndim(state)
+
+    if n_dim >= TENSORDOT_STATE_NDIM_PERF_THRESHOLD and math.get_interface(state) == "tensorflow":
+        return apply_operation_tensordot(op, state, is_state_batched=is_state_batched)
+
+    # Common constants always needed
+    n_dim = math.ndim(state)
+    num_wires = _get_num_wires(state, is_state_batched)
+
+    # Start applying from the left side
+    axis = op.wires[0] + is_state_batched
+
+    # Slice indices of the affected axis
+    sl_0 = _get_slice(0, axis, n_dim)
+    sl_1 = _get_slice(1, axis, n_dim)
+
+    # Get the phase shift parameter
+    params = math.cast(op.parameters[0], dtype=complex)
+    state0 = state[sl_0]
+    state1 = state[sl_1]
+    if op.batch_size is not None and len(params) > 1:
+        interface = math.get_interface(state)
+        if interface == "torch":
+            params = math.array(params, like=interface)
+        if is_state_batched:
+            # If both op and state are batched, they have to have the same batch size
+            params = math.reshape(params, (-1,) + (1,) * (n_dim - 2))
+        else:
+            # Op is batched, state is not, so we need to expand the state to batched
+            params = math.reshape(params, (-1,) + (1,) * (n_dim - 1))
+            state0 = math.expand_dims(state0, 0) + math.zeros_like(params)
+            state1 = math.expand_dims(state1, 0)
+            # Update status
+            is_state_batched = True
+            axis = axis + 1
+            n_dim = n_dim + 1
+    state1 = math.multiply(math.cast(state1, dtype=complex), math.exp(1.0j * params))
+    state = math.stack([state0, state1], axis=axis)
+    # Left side finished
+
+    # Now start right side
+    axis += num_wires  # Move to the right side (conjugate side)
+    # Slice indices of the affected axis
+    sl_0 = _get_slice(0, axis, n_dim)
+    sl_1 = _get_slice(1, axis, n_dim)
+    # Get the phase shift parameter, conjugated
+    state0 = state[sl_0]
+    state1 = state[sl_1]
+    # No need for expanding, since on the left side we already did
+
+    state1 = math.multiply(math.cast(state1, dtype=complex), math.exp(-1.0j * params))
+    state = math.stack([state0, state1], axis=axis)
+    return state
+
+
+@apply_operation.register
```

Comment:
```suggestion
@apply_operation.register(qml.CNOT)
@apply_operation.register(qml.MultiControlledX)
@apply_operation.register(qml.Toffoli)
```

With `single_dispatch`, we can have the same function registered to multiple different types.  Might help cut down on some duplicated code.

Does this formalism hold for any self-inverse operation, or just the controlled-x variants? MIght it also provide a performance boost for `SWAP`, `CSWAP`, `CZ`, `CY`, `CH`, etc? 

Since running the performance numbers takes some time though, I'm ok with not adding in this kernel for more operators.